### DISCUSSION TITLE
Abandonner l'attribut "références"

### DISCUSSION THREAD

**Discussion by laem** [publicodes/publicodes#198] [Ideas]
Je me rend comte que lister des références sans une note qui précise quelle page dans un PDF, ou autres informations contextuelles utiles, n'est pas une très bonne idée.

Je me retrouve souvent à préciser tout ça dans le champs `note` ou `description`.

Je propose donc de supprimer ce champs, et de faire en sorte que les pages de documentation compilent automatiquement les url présentes dans le markdown en bas de page.

Autre solution : utiliser un plugin markdown qui permet d'intégrer des notes de bas de page (je l'ai intégré sur nosgestesclimat).

**Comment by mquandalle** [comment]
> sans une note qui précise quelle page dans un PDF

Sur ce point précis tu peux ajouter une ancre `#page=30` qui est supporté par la plupart des clients PDF [exemple](https://minefi.hosting.augure.com/Augure_Minefi/r/ContenuEnLigne/Download?id=0638C996-A2C6-48E5-86FB-A53A22A780F1&filename=DP%20-%20PLF%202021.pdf#page=34)

**Comment by laem** [comment]
Merci ! Je savais pas ça.

**Comment by mquandalle** [comment]
Sinon c'est une idée intéressante de regrouper les références et notes de bas de page dans un champ description en markdown.

Peut-être même qu'on pourrait aller plus loin et passer à un modèle de « [programmation littéraire](https://fr.wikipedia.org/wiki/Programmation_lettr%C3%A9e) » où on utiliserait un format de premier niveau compatible Markdown et où le code serait dans des section indentées : [exemple sur le Publicodes entreprise](https://gist.github.com/mquandalle/d87d21290ea15b621f041c19492a389a). On mettrait ainsi en avant la documentation (incluant références et notes de bas de page donc) dans le code source et sur GitHub. Le code serait dans les sections indentées (mais c'est déjà le cas aujourd'hui car avec le yaml on doit indenter le contenu de la règle, alors qu'avec un titre markdown on peut écrire en dessous sans indentation).

[Catala](https://catala-lang.org/) utilise aussi la programmation littéraire, j'ai [ouvert une discussion sur leur Zulip](https://zulip.catala-lang.org/#narrow/stream/1-general/topic/Markdown.20compatible.20literate.20programming) pour avoir leur avis sur une syntaxe littérale compatible avec Markdown.

À réfléchir, mais je voulais partager la réflexion :)

**Comment by johangirod** [comment]
Passer du yaml au md, ambitieux :)
J'ai beaucoup. Mais il faudrait voir à l'usage. 

On pourrait même imaginer d'utiliser [les niveaux de titre comme sémantique d'espace de nom](https://gist.github.com/johangirod/d2c096cee7d088d86c969388d7fba930).

**Comment by laem** [comment]
C'est très intéressant en effet. Je me demande si on pourrait pas faire coexister facilement deux types de fichiers sources, en fonction des préférences ou des spécificités (volume de doc par exemple) du domaine implémenté.

**Comment by mquandalle** [comment]
Je reviens sur l'idée de passer à un format markdown plutôt que yaml comme langage de “premier niveau”. En effet j'ai l'impression que ça faciliterait l'appropriation du langage par des non-programmeurs. Pour l'anecdote la réflexion m'est venue en regardant @penelopel créer et modifier des documents markdown avec grande aisance (😋).

Un format dérivé du markdown aurait aussi l'avantage d'être une représentation texte proche de l'affichage de la règle dans la documentation — plus que le yaml. Dans le bac à sable, la vue en deux colonnes serait une transposition directe entre les deux représentations.

J'ai l'impression aussi que le markdown apparaît de plus en plus dans des plateformes “no-code”, par exemple sur Airtable on peut l'utiliser pour [formater du texte](https://support.airtable.com/hc/en-us/articles/360044741993-Markdown-syntax-for-Airtable-rich-text-formatting), et à côté de ça on a un [langage de formules](https://support.airtable.com/hc/en-us/articles/203255215-Formula-field-reference) à la Excel/SQL. 

Comme Airtable on souhaite un outil utilisable par le plus grand nombre, et par ailleurs on utilise déjà du markdown pour le formatage des textes et un langage de formules pour les calculs. Peut-être donc que l'utilisation du yaml comme langage de “premier niveau” n'est pas essentielle et que l'on peut s'en passer ?

Je pense creuser la question et expérimenter un nouveau format sur une partie de nos règles. Bien sûr si on va au bout il nous faudra des scripts de migrations, etc., mais voyons d'abord si un tel changement vaut le coup.

**Comment by laem** [comment]
Je confirme mon étonnement sur la capacité des non-dev que je côtoie à maîtriser le markdown :) 

:+1: pour l'expérimentation. Par contre, peut-être créer une autre issue ? Je pense que ma 1ère demande reste valide et est simple à implémenter. 

Surtout qu'il y a des pseudos standards aujourd'hui pour gérer les notes de bas de page en markdown.
